homeassistant/components/media_player/gpmdp.py

```diff
@@ -0,0 +1,123 @@
+        return False
+
+    add_devices([GPMDP(name, address, create_connection, json)])
+
+
+class GPMDP(MediaPlayerDevice):
+    """Representation of a GPMDP."""
+
+    # pylint: disable=too-many-public-methods, abstract-method
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, name, address, create_connection, json):
+        """Initialize."""
+        self._connection = create_connection
+        self._json = json
+        self._address = address
+        self._name = name
+        self._status = STATE_OFF
+        self.update()
+
+    def update(self):
+        """Get the latest details from the player."""
+        import socket
+        try:
+            self._ws = self._connection(("ws://" + self._address + ":5672"),
+                                        timeout=1)
+            state = self._ws.recv()
+            if ((self._json.loads(state))['payload']) is True:
+                song = self._ws.recv()
+                self._title = ((self._json.loads(song))['payload']['title'])
+                self._artist = ((self._json.loads(song))['payload']['artist'])
+                self._albumart = ((self._json.loads(song))['payload']
+                                  ['albumArt'])
+                self._status = STATE_PLAYING
+            elif ((self._json.loads(state))['payload']) is False:
+                self._status = STATE_PAUSED
+        except (socket.timeout, ConnectionRefusedError):
+            self._status = STATE_OFF
+
+    @property
+    def media_content_type(self):
+        """Content type of current playing media."""
+        return MEDIA_TYPE_MUSIC
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._status
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self._title
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media (Music track only)."""
+        return self._artist
+
+    @property
+    def media_image_url(self):
+        """Image url of current playing media."""
+        return self._albumart
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def supported_media_commands(self):
+        """Flag of media commands that are supported."""
+        return SUPPORT_GPMDP
+
+    def media_next_track(self):
+        """Send media_next command to media player."""
+        self._ws.send('{"namespace": "playback", "method": "forward"}')
+        self.update_ha_state()
+
+    def media_previous_track(self):
+        """Send media_previous command media player."""
+        self._ws.send('{"namespace": "playback", "method": "rewind"}')
+        self.update_ha_state()
+
+    def media_play(self):
+        """Send media_play command to media player."""
+        self._ws.send('{"namespace": "playback", "method": "playPause"}')
+        self._status = STATE_PLAYING
```

Comment:
Why are you doing an optimistic update, is the new state not reported quick enough ?